### PR TITLE
Fixed adding of points when a goal is marked completed

### PR DESCRIPTION
**Pull request by buchii1**
I removed the functionality that adds points to **SimpleGoal**, **ProgressionGoal**, and **ChecklistGoal** when a user completes either of the goal. 
I also made use of `do...while` loop in implementing minor error handling logic that ensures that the program doesn't break when a user inputs/selects the wrong choice/option.

### CODE REVIEW

File: prove/Develop05/ChecklistGoal.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public override bool CheckCompletionStatus()
 
     public override void DisplayProgressMessage()
     {
-        if (CheckCompletionStatus())
+        if (_numOfTimesAccomplished == _duration)
```
Comment (nvparrish) [inline prove/Develop05/ChecklistGoal.cs:27]: I would normally use an inequality here.  Checking for equality can bite you if for some reason the count jumps.

File: prove/Develop05/GoalTracker.cs [hunk 2]
```diff
@@ -33,13 +34,21 @@ public void AddGoal(Goal goal)
 
     public void DisplayGoals()
     {
-        Console.WriteLine("The goals are:");
+        if (_goals.Count != 0)
+        {
+            Console.WriteLine("The goals are:");
 
-        for (int i = 0; i < _goals.Count; i++)
+            for (int i = 0; i < _goals.Count; i++)
```
Comment (nvparrish) [inline prove/Develop05/GoalTracker.cs]: I like the foreach for these types of structures when you are iterating through a list of things.

File: prove/Develop05/Program.cs [hunk 3]
```diff
@@ -22,41 +22,47 @@ static void Main(string[] args)
...
-                    case 3:
-                        checklistGoal.DisplayStartMessage();
-                        trackGoal.AddGoal(checklistGoal);
-                        break;
-                    default:
-                        progressionGoal.DisplayStartMessage();
-                        trackGoal.AddGoal(progressionGoal);
-                        break;
-                }
+                    trackGoal.DisplayGoalTypes();
```
Comment (nvparrish) [inline prove/Develop05/Program.cs]: This structure doesn't allow you to exit the part where you are adding goals.

File: prove/Develop05/ProgressionGoal.cs [hunk 4]
```diff
@@ -44,20 +52,28 @@ public override bool CheckCompletionStatus()
...
+                Spin();
+            }
+
+            if (CheckCompletionStatus())
+            {
+                _totPoint = 0; // Reset value to zero
+                Console.WriteLine($"You have completed all {_totalLevel} levels of the {_name} goal. You rock!");
+                _totPoint = _bonusPoints + _completionBonus;
+                _progress = (_duration += 1);
+                Spin();
```
Comment (nvparrish) [inline prove/Develop05/ProgressionGoal.cs:75]: I don't love the spinner here.  It makes things take longer for testing.

File: prove/Develop05/GoalTracker.cs (hunk 2 above)
Comment (buchii1) [reply prove/Develop05/GoalTracker.cs]: Fixed

File: prove/Develop05/Program.cs (hunk 3 above)
Comment (buchii1) [reply prove/Develop05/Program.cs]: Fixed

File: prove/Develop05/ProgressionGoal.cs (hunk 4 above)
Comment (buchii1) [reply prove/Develop05/ProgressionGoal.cs:75]: Spinner duration adjusted

File: prove/Develop05/ChecklistGoal.cs (hunk 1 above)
Comment (buchii1) [reply prove/Develop05/ChecklistGoal.cs:27]: Will be worked on later.
Thanks

**Pull request merged**